tests/app/views/contexts/summary/test_question.py

```diff
@@ -77,6 +78,126 @@ def get_question_schema(answer_schema):
+                                                {
+                                                    "id": "address-line-3",
+                                                    "label": "Address Line 3",
+                                                    "mandatory": False,
+                                                    "type": "TextField",
+                                                },
+                                                {
+                                                    "id": "town-city",
+                                                    "label": "Town/City",
+                                                    "mandatory": False,
+                                                    "type": "TextField",
+                                                },
+                                                {
+                                                    "id": "county",
+                                                    "label": "County",
+                                                    "mandatory": False,
+                                                    "type": "TextField",
+                                                },
+                                                {
+                                                    "id": "postcode",
+                                                    "label": "Postcode",
+                                                    "mandatory": False,
+                                                    "type": "TextField",
+                                                },
+                                                {
+                                                    "id": "country",
+                                                    "label": "Country",
+                                                    "mandatory": False,
+                                                    "type": "TextField",
+                                                },
+                                            ],
+                                        },
+                                    },
+                                ],
+                                "id": "address-group",
+                            }
+                        ],
+                    }
+                ]
+            }
+        )
+
+    def address_question(self, concatenation_type):
+
+        answers = [
+            self.address_questionnaire_schema().get_answers_by_answer_id(answer_id)[0]
+            for answer_id in self.address_questionnaire_schema().get_answer_ids_for_block(
+                "what-is-your-address"
+            )
+        ]
+
+        question_schema = {
+            "id": "question_id",
+            "title": "question_title",
+            "type": "General",
+            "answers": answers,
+            "summary": {"concatenation_type": concatenation_type},
+        }
+
+        return Question(
+            question_schema,
```

Comment:
You basically already have a full schema, so I would update `address_questionnaire_schema` to take `concatenation_type` as a param, then this can be simplied to:
```py
  question_schema = self.schema.get_questions("what-is-your-address-question")[0]
  return Question(
     question_schema,
```

You will have to move `self.schem` to inside the subtest loop and pass in the concat.
```
self.schema = self.address_questionnaire_schema(concatenation_type)
```

Reply:
Ok, added all your suggestions to the test, all simplified and working good. Thanks.